### ISSUE TITLE
Highcharts Editor Replacement

### ISSUE THREAD

**Issue by dan-bowerman** [ramp4-pcar4/storylines-editor#371]
High-level requirements:
- [ ] Chart preview
- [ ] Paste in data
- [ ] Import CSV
- [ ] *Develop English and French simultaneously (like RAMP4 editor)
- [ ] Import existing Highcharts JSON configuration
- [ ] Export as Highcharts JSON configuration
- [ ] Chart title
- [ ] Chart sub-title
- [ ] Data Series configs: Color, color by point, dash style, enable point markers?, marker symbol
- [ ] Y-axis title
- [ ] X-axis title (if needed)

Chart/Series types - Keep it basic on the first pass, add more types as demand dictates
- [ ] Bar
- [ ] Column
- [ ] Line
- [ ] Scatter
- [ ] *Combinations of the above
- [ ] Pie

Things we could hardcode/make default:
- [ ] *Filename
- [ ] Width, height, zoom: auto
- [ ] Fonts and styles
- [ ] Value labels
- [ ] Legends
- [ ] Number formatting
- [ ] Export button and menu (localization)
- [ ] Disable credits

Other considerations:
- An "Advanced" config editor to allow straight JSON manipulation?
- This may want its own repo? Do we want it to be standalone outside of RESPECT?
- Look and feel should be similar to the RAMP4 config-editor, but consultation with UX Team is advised.
- Should be fully WCAG "AA" compliant.

`* Needs further discussion`

**Comment by yileifeng** [comment]
List looks good to me and includes all of the core functionality. Having an advanced editor (similar to what we have for slides) seems like a good idea to cover any gaps from missing advanced functionality such as secondary axis positioning, tooltip/legend formatting, etc. The consideration for integrating an advanced editor would involve drafting up a comprehensive Highcharts schema for validation.

**Comment by james-rae** [comment]
> This may want its own repo? Do we want it to be standalone outside of RESPECT?

I remember someone mentioning in a meeting that general real-world users of the current highcharts editor were awfully unhappy due to the lack of support.

Having this as a stand-alone might have a lot of use for a lot of people (and maybe even a few of them discover their StoryRamp dreams can come true). On the flipside, if this does become popular, one can expect all sorts of issues being logged, requests for new features, and the obligatory grouses and unpleasantness of anything public. I can't see many asks getting prioritized unless it's something also benefiting our projects.

Anyway, could be..._interesting_.

**Comment by abooalir** [comment]
Hello everyone. We have some questions regarding the Highcharts requirements:

- For the chart types, are you looking to only have the basic option for each type, or do you also want us to consider the different styles of that chart similar to Highcharts?
![Screenshot 2024-09-20 133249](https://github.com/user-attachments/assets/db2449e2-4b8a-436d-81d0-d70e31d978b1)
- With regards to importing Highcharts as JSON configuration, will the user be allowed to upload a .JSON file or copy and paste into an editor or both?
 -Just to confirm, if the user uploads a new data file, is the expected behaviour to replace any existing imported data?
- What are the formats the user can export the data as?

We would really appreciate your clarification on these matters. Thank you in advance.

**Comment by dan-bowerman** [comment]
> For the chart types, are you looking to only have the basic option for each type, or do you also want us to consider the different styles of that chart similar to Highcharts?

I think just the basics for this first phase. We can get very bogged down with supporting the vast selection of chart types and options for them, but at the end of the day, 99% of the charts our clients are going to be developing will be basic, or combinations of basic types. Folks will be able to manually construct and import "fancy" charts using the other chart types available if they desire. 

> With regards to importing Highcharts as JSON configuration, will the user be allowed to upload a .JSON file or copy and paste into an editor or both?

Both would be preferred. To my point above, a client may use the Highcharts editor on the side if they so desire, and they could import the file exported from that tool, or paste it in directly. 

> Just to confirm, if the user uploads a new data file, is the expected behaviour to replace any existing imported data?

The Storylines Editor's current implementation requires a user to create a blank chart, and from there they can work with the H.E.R. charts editor directly, or import an existing configuration. Given the context they are working in, I believe it makes sense to overwrite that selected chart if they're importing a configuration. 

> What are the formats the user can export the data as?

Export in this context means to output the generated JSON output of the Highcharts Editor such that it could be leveraged in other Storylines products or the original Highcharts Editor. It would be a .JSON file that is downloaded to the user's device. 

Hope this clarifies things, please reach out if you have any more questions!
